### PR TITLE
fix(page): Page.createIsolatedWorld error catching has been added

### PR DESCRIPTION
**Pull request by DimaShustal**
**What kind of change does this PR introduce?**
Bug fix

**Did you add tests for your changes?**
No

**If relevant, did you update the documentation?**
Not relevant

**Summary**
When navigating, frames are destroyed and sending events leads to an error that crashes the whole process. This error was mentioned [here](https://github.com/puppeteer/puppeteer/pull/4707). Error example:
```
ProtocolError: Protocol error (Page.createIsolatedWorld): No frame for given id found
at /.../node_modules/puppeteer/src/common/Connection.ts:300:16
at new Promise (<anonymous>)
at CDPSession.send (/.../node_modules/puppeteer/src/common/Connection.ts:296:12)
at /.../node_modules/puppeteer/src/common/FrameManager.ts:397:19
at Array.map (<anonymous>)
at FrameManager._ensureIsolatedWorld (/.../node_modules/puppeteer/src/common/FrameManager.ts:396:10)
at async Promise.all (index 1)
at async FrameManager.initialize (/.../node_modules/puppeteer/src/common/FrameManager.ts:143:7)
at async FrameManager._onAttachedToTarget (/.../node_modules/puppeteer/src/common/FrameManager.ts:262:5) {
originalMessage: 'No frame for given id found'
}
```

**Does this PR introduce a breaking change?**
As I can see there is no breaking change

### CODE REVIEW

File: src/common/FrameManager.ts [hunk 1]
```diff
@@ -398,7 +398,7 @@ export class FrameManager extends EventEmitter {
             frameId: frame._id,
             worldName: name,
             grantUniveralAccess: true,
-          })
+          }).catch(console.error)
```
Comment (eemelipa) [inline src/common/FrameManager.ts]: Earlier this was 

```suggestion
          }).catch(debugError)
```

imported with `import {debugError} from './helper';`. Should it be the debugError instead?

(For some reason it got removed in https://github.com/puppeteer/puppeteer/commit/4d9dc8c0e613f22d4cdf237e8bd0b0da3c588edb)

Comment (eemelipa) [reply src/common/FrameManager.ts]: 👍 

Looks good now

**Comment by eemelipa** [comment]
LGTM

I tried the fix in my use case and it seemed to work as expected

Someone from Puppeteer team should give their blessing as well. I don't know all the implications of the change, and why it broke now. I believe the catch clause hasn't been there for a few puppeteer releases so some other change has made this issue surface again

**Review by eemelipa — APPROVED** [review]
LGTM

I tried the fix in my use case and it seemed to work as expected

Someone from Puppeteer team should give their blessing as well. I don't know all the implications of the change, and why it broke now. I believe the catch clause hasn't been there for a few puppeteer releases so some other change has made this issue surface again (e.g. with puppeteer 11.0.0 it wasn't there but v11 worked ok for me)

**Comment by OrKoN** [comment]
Hey, thanks for the PR. Could you please sign a CLA and rebase (with code style fixes `npm run eslint-fix`) to get this landed?

**Comment by OrKoN** [comment]
Didn't mean to close :D

**Review by OrKoN — CHANGES_REQUESTED** [review]
marking as requiring changes (see my previous comment)

**Comment by DimaShustal** [comment]
I will update PR tomorrow

**Comment by dmyerscough** [comment]
When will this change make it into Puppeteer?

**Review by OrKoN — APPROVED** [review]

**Pull request merged**